### PR TITLE
Epymarl wrapper

### PR DESCRIPTION
**Pull request by moratodpg**
- Created a wrapper for EPyMARL inheriting from PyMARL ma

### CODE REVIEW

File: imp_marl/imp_wrappers/examples/epymarl/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- Additional algorithms (IA2C, IPPO, MAPPO and MAA2C)
+- Option for no-parameter sharing between agents
+- Flexibility with extra implementation details
+
+For more information, check their [blog](https://agents.inf.ed.ac.uk/blog/epymarl/)
+
+You can run IMP-MARL environments on EPyMARL by:
+
+1. Create a new virtual environment.
+2. Install imp_marl package: `pip install imp-marl`.
```
Comment (PaLeroy) [inline imp_marl/imp_wrappers/examples/epymarl/README.md]: This does not exist yet

File: imp_marl/imp_wrappers/README.md [hunk 2]
```diff
@@ -13,12 +13,14 @@ Each wrapper has its own requirements. Please check the import of each wrapper,
 ## Available wrappers:
 - [PymarlMAStruct](pymarl_wrapper/pymarl_wrap_ma_struct.py): Multi-agent wrapper for the [MultiAgentEnv](pymarl_wrapper/MultiAgentEnv.py) interface required by [PyMarl](pymarl_wrapper/README.md).
 - [PymarlSAStruct](pymarl_wrapper/pymarl_wrap_sa_struct.py): Single-agent wrapper for the [MultiAgentEnv](SingleAgentEnv.py) interface required by [PyMarl](pymarl_wrapper/README.md).
+- [EPymarlMAStruct](epymarl_wrapper/epymarl_wrap_ma_struct.py): Multi-agent wrapper for the [MultiAgentEnv](pymarl_wrapper/MultiAgentEnv.py) interface required by [EPyMarl](https://github.com/uoe-agents/epymarl/tree/main).
```
Comment (PaLeroy) [inline imp_marl/imp_wrappers/README.md]: Is the second link is correct?

File: imp_marl/imp_wrappers/examples/epymarl/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+3. [Clone EPyMARL](https://github.com/uoe-agents/epymarl).
+4. Install the required EPyMARL modules according to the requirements file: `pip install -r requirements`.
+5. Register your imp_marl env(s) within EPyMARL. Do that on `/src/envs/__init__.py`:
+    ```
+    from imp_marl.imp_wrappers.epymarl_wrapper.epymarl_wrap_ma_struct import EPymarlMAStruct
+    def env_fn(env, **kwargs) -> MultiAgentEnv:
+    return env(**kwargs)
+    REGISTRY = {}
+    REGISTRY["struct_marl"] = partial(env_fn, env=EPymarlMAStruct)
```
Comment (PaLeroy) [inline imp_marl/imp_wrappers/examples/epymarl/README.md:20]: I think only the first and last lines are not already in the file and one should not overwrite the others?

File: imp_marl/imp_wrappers/examples/epymarl/README.md (hunk 1 above)
Comment (PaLeroy) [reply imp_marl/imp_wrappers/examples/epymarl/README.md]: Maybe stick with `pip install git+http://github.com/moratodpg/imp_marl.git` for now?

File: imp_marl/imp_wrappers/examples/epymarl/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    ```
+6. Include the config files associated with your env(s) in `/src/config/envs`. For instance, you can include `struct_uc_3.yaml` config file.
```
Comment (PaLeroy) [inline imp_marl/imp_wrappers/examples/epymarl/README.md]: These files are now hidden in `papers/neurips_23/pymarl/config/envs/struct_c_3.yaml`, maybe we add a hint to help people find them?

Comment (PaLeroy) [reply imp_marl/imp_wrappers/examples/epymarl/README.md]: You need also to add a map_name in the config file.
Maybe we can provide one config file in this directory as an example?

**Review by PaLeroy — COMMENTED** [review]
Small things to address.
Anyway, I was able to make it run easily based on the guidelines! 
Nice work 🥳

**Comment by PaLeroy** [comment]
Also added the lint check 😄

File: imp_marl/imp_wrappers/examples/epymarl/README.md (hunk 1 above)
Comment (moratodpg) [reply imp_marl/imp_wrappers/examples/epymarl/README.md]: Agreed!

File: imp_marl/imp_wrappers/README.md (hunk 2 above)
Comment (moratodpg) [reply imp_marl/imp_wrappers/README.md]: Technically, it is also derived from pymarl main env template. But I got your point - let's remove the ref here.

**Comment by moratodpg** [comment]
Thanks for the remarks! I have now addressed all of them.
Are we good to go for the merge? :)
It seems the checks triggered by actions have failed xD

**Comment by PaLeroy** [comment]
I think this PR should be reopen to address remaining remarks.

**Pull request merged**